Sources/MlemMiddleware/Content Models/Post/PostStubProviding.swift

```diff
@@ -64,3 +64,12 @@ public extension PostStubProviding {
     var read_: Bool? { nil }
     var myVote_: ScoringOperation? { nil }
 }
+
+public extension PostStubProviding {
+    func upgrade() async throws -> Post2 {
```

Comment:
In the community upgrade system, `upgrade` is overridden in `Community3` to simply return `self`:

```swift
// Community3.swift, line 36
public func upgrade() async throws -> Community3 { self }
```
 This avoids making requests when calling `upgrade` on an already upgraded model. For the override to work, `upgrade()` is explicitly required in the `CommunityStubProviding` definition.
 
 This implementation should be mirrored for the post upgrade system.